File: frontend/netrics/src/app/service/pulse/standard-raw-kpi-mapping-service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {UrlService} from '../url/url-service';
import {Observable} from 'rxjs';
import {StandardRawKpiMappingDto} from '../../models/pulse/StandardRawKpiMappingDto';
import {StandardKpiDto} from '../../models/pulse/StandardKpiDto';

@Injectable({
  providedIn: 'root'
})
export class StandardRawKpiMappingService {

  private readonly baseUrl: string;

  constructor(private http: HttpClient, private urlService: UrlService) {
    this.baseUrl = `${this.urlService.getPulseUrl()}/standard-raw-kpimap`;
  }

  getAll(ratName: string): Observable<StandardRawKpiMappingDto[]> {
    return this.http.get<Array<StandardRawKpiMappingDto>>(this.baseUrl, { params: { ratName } });
  }

  createMapping(dto: StandardRawKpiMappingDto): Observable<StandardRawKpiMappingDto> {
    return this.http.post<StandardRawKpiMappingDto>(this.baseUrl, dto);
  }

  isMappingAvailable(ratName: string, standardKpiName: string): Observable<boolean> {
    return this.http.get<boolean>(`${this.baseUrl}/is-available`, {
      params: {
        ratName: ratName,
        standardKpiName: standardKpiName
      }
    });
  }

}

File: backend/spring/pulse/src/main/java/dev/thilanka/netrics/service/StandardRawKpiMappingService.java
package dev.thilanka.netrics.service;

import dev.thilanka.netrics.dto.StandardRawKpiMappingDto;
import jakarta.validation.Valid;

import java.util.List;

public interface StandardRawKpiMappingService {
    List<StandardRawKpiMappingDto> getAll(String ratName);

    StandardRawKpiMappingDto createMapping(StandardRawKpiMappingDto dto);

    List<StandardRawKpiMappingDto> createMappingList(@Valid List<StandardRawKpiMappingDto> dtos);

    boolean isKpiMappingAvailable(String ratName, String standardKpiName);
}
